src/main.js

```diff
@@ -1,33 +1,98 @@
 saveButton.addEventListener('click', saveCard)
 titleInput.addEventListener('keyup', enableSave)
 bodyInput.addEventListener('keyup', enableSave)
-window.addEventListener('load', loadFromStorage)
+searchButton.addEventListener('click', searchAndClear)
+searchBar.addEventListener('keyup', search)
 ideasGrid.addEventListener('click', runStarBar)
 
 
-// variables
+//functions
 
-var cards = []
+/*creates a new array to hold array of starred cards
+if the button says 'Show Starred Ideas' when clicked, it changes to 'Show All Ideas'
+loops through the cards and checks if the card's 'star' property is true
+if it's starred, that Idea instance gets added to the array
+that starred array is displayed
+if the button says 'Show All ideas' when clicked, it changes the button to say "Show Starred Ideas"
+it displays all the idea cards
+it clears the starredCards array so as to avoid duplication next time the starred button is clicked
+*/
+function showStarredCards() {
+  var starredCards = []
+  if (showStarredButton.innerText === 'Show Starred Ideas') {
+    showStarredButton.innerText = 'Show All Ideas'
+    for (var i = 0; i < cards.length; i++) {
+      if (cards[i].star) {
+        starredCards.push(cards[i])
+      }
+    }
+    displayCards(starredCards)
+  } else {
+    showStarredButton.innerText = 'Show Starred Ideas'
+    displayCards(cards)
+    starredCards = []
+  }
+}
 
-//functions
+/*
+the preventDefault stops the page from just reloading to display all cards
+this declares a new array to hold our search searchResults  >>>>> searchResults
+grabs the input from the user  >>>>> userSearch
+this goes through each card in our cards array >>>>> for... loop
+calls a method on the card that checks if the search is in that card's body or title >>> .checkIfContains()
+gets the boolean saying if it does >>> if cards[i].contains
+pushes the results into a searchResults array >>>>> searchResults.push(cards[i])
+and then displays it >>>>> displayCards()
+we also clear the searchResults array to be clear for next search
+
+NOT WORKING PROPERLY I think because of the method? I logged the input and the array, and nothing is getting added to the results array but somehow some stuff is still getting filtered out, bewilderingly.
+If you do a search and then delete and do another, nothing happens.
+*/
+function search(event) {
+  event.preventDefault()
+  var searchResults = []
+
+  var userSearch = searchBar.value
+// console.log('Search input:', userSearch, 'Results Array:', searchResults)
+// debugger
+  for (var i = 0; i < cards.length; i++) {
+    cards[i].checkIfContains(userSearch)
+    if (cards[i].containsSearch) {
+      searchResults.push(cards[i])
+    }
+  }
 
-function saveCard(event) {
+  displayCards(searchResults)
+  clearResults(searchResults)
+  // searchResults = []
+}
+
+function clearResults(resultsArray) {
+  for (var i = 0; i < resultsArray.length; i++) {
+    resultsArray[i].contains = false
```

Comment:
Seems like you accidentally a word here. Should say `resultsArray[i].containsSearch = false` right?